src/Text/Pandoc/Readers/JATS.hs

```diff
@@ -321,24 +352,28 @@ parseBlock (Elem e) =
                                      (TableFoot nullAttr [])
          isEntry x  = named "entry" x || named "td" x || named "th" x
          parseElement = filterChildren isEntry
-         sect n = do isbook <- gets jatsBook
-                     let n' = if isbook || n == 0 then n + 1 else n
-                     labelText <- case filterChild (named "label") e of
-                                    Just t -> (<> ("." <> space)) <$>
-                                              getInlines t
-                                    Nothing -> return mempty
-                     headerText <- case filterChild (named "title") e `mplus`
-                                        (filterChild (named "info") e >>=
-                                            filterChild (named "title")) of
-                                      Just t  -> (labelText <>) <$>
-                                                  getInlines t
-                                      Nothing -> return mempty
-                     oldN <- gets jatsSectionLevel
-                     modify $ \st -> st{ jatsSectionLevel = n }
-                     b <- getBlocks e
-                     let ident = attrValue "id" e
-                     modify $ \st -> st{ jatsSectionLevel = oldN }
-                     return $ headerWith (ident,[],[]) n' headerText <> b
+         parseWithHeader n mBlocks = do 
+                            isbook <- gets jatsBook
+                            let n' = if isbook || n == 0 then n + 1 else n
+                            labelText <- case filterChild (named "label") e of
+                                            Just t -> (<> ("." <> space)) <$>
+                                                      getInlines t
+                                            Nothing -> return mempty
+                            headerText <- case filterChild (named "title") e `mplus`
+                                                (filterChild (named "info") e >>=
+                                                    filterChild (named "title")) of
+                                              Just t  -> (labelText <>) <$>
+                                                          getInlines t
+                                              Nothing -> return labelText
+                            oldN <- gets jatsSectionLevel
+                            modify $ \st -> st{ jatsSectionLevel = n }
+                            blocks <- mBlocks
+                            let ident = attrValue "id" e
+                            modify $ \st -> st{ jatsSectionLevel = oldN }
+                            return $ (if 
+                              headerText == mempty 
+                            then mempty 
+                            else headerWith (ident,[],[]) n' headerText) <> blocks
```

Comment:
This is also something I wasn't sure about, but seemed wrong. Previously, we would include a header block even if there was no content to put inside of it (`headerText == mempty`). Without this `if` check, every instance of all the tags that use this function will include a header block even if there isn't any content to put inside of it

Reply:
Looks correct to me. I do not know if there is any rule in ATL for keeping empty headers, if there is not, then I would say this is a good solution.